proposals/numeric-intptr.md

```diff
@@ -0,0 +1,307 @@
+# Numeric IntPtr
+
+## Summary
+[summary]: #summary
+
+This is a revision on the initial native integers feature ([spec](https://github.com/dotnet/csharplang/blob/main/proposals/csharp-9.0/native-integers.md)), where the `nint`/`nuint` types were distinct from the underlying types `System.IntPtr`/`System.UIntPtr`.
+In short, we now treat `nint`/`nuint` as simple types aliasing `System.IntPtr`/`System.UIntPtr`, like we do for `int` in relation to `System.Int32`.
+
+## Design
+[design]: #design
+
+### 8.3.5 Simple types
+
+C# provides a set of predefined `struct` types called the simple types. The simple types are identified through keywords, but these keywords are simply aliases for predefined `struct` types in the `System` namespace, as described in the table below.
+
+**Keyword** | **Aliased type**
+----------- | ------------------
+  `sbyte`   |   `System.SByte`
+  `byte`    |   `System.Byte`
+  `short`   |   `System.Int16`
+  `ushort`  |   `System.UInt16`
+  `int`     |   `System.Int32`
+  `uint`    |   `System.UInt32`
+  **`nint`**     |   **`System.IntPtr`**
+  **`nuint`**    |   **`System.UIntPtr`**
```

Comment:
IMO, its the same consideration as `Int32` vs `int` and the language keyword should be preferred.

There are likely some devs who may argue that `IntPtr` should still be used when you are directly dealing with an `opaque handle` (like `System.Windows.Forms.Form.Handle`), but from the IDE, docs, and other perspectives you are just going to see/get `nint`.

Reply:
@BillWagner Tooling will follow the same rules as for other special types with regards to preferring keyword over full-type-name. So I assume that other style guidelines will follow the same model.
As Tanner pointed out, it will become a bit harder to say `System.IntPtr` (to mean "opaque handle") as opposed to `nint`...